src/Admin/BaseBlockAdmin.php

```diff
@@ -163,7 +198,14 @@ public function preRemove($object)
      */
     public function postRemove($object)
     {
-        $this->blockManager->get($object)->postRemove($object);
+        if (\is_callable([$object, 'postRemove'])) {
+            $this->blockManager->get($object)->postRemove($object);
+
+            @trigger_error(
+                'The '.__METHOD__.' method is deprecated since version 3.x and will be removed in sonata/block-bundle 4.0.',
```

Comment:
```suggestion
                'The '.__METHOD__.'() method is deprecated since sonata-project/block-bundle 3.x and will be removed in version 4.0.',
```